File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/[VB]-Visual Studio 2010 SDK Samples/Code Sweep/C++,VB/VsPackage/IBackgroundScanner.vb
'**************************************************************************

'Copyright (c) Microsoft Corporation. All rights reserved.
'This code is licensed under the Visual Studio SDK license terms.
'THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
'ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
'IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
'PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

'**************************************************************************


Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports Microsoft.VisualStudio.Shell.Interop

Namespace Microsoft.Samples.VisualStudio.CodeSweep.VSPackage
    Friend Interface IBackgroundScanner
        Event Started As EventHandler
        Event Stopped As EventHandler

        ReadOnly Property IsRunning() As Boolean
        Sub Start(ByVal projects As IEnumerable(Of IVsProject))
        Sub RepeatLast()
        Sub StopIfRunning(ByVal blockUntilDone As Boolean)
    End Interface
End Namespace

File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/60968-Visual Studio 2010 SDK Samples/Code Sweep/C#,C++/VsPackage/IBackgroundScanner.cs
/***************************************************************************

Copyright (c) Microsoft Corporation. All rights reserved.
This code is licensed under the Visual Studio SDK license terms.
THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

***************************************************************************/

using System;
using System.Collections.Generic;
using Microsoft.VisualStudio.Shell.Interop;

namespace Microsoft.Samples.VisualStudio.CodeSweep.VSPackage
{
    interface IBackgroundScanner
    {
        event EventHandler Started;
        event EventHandler Stopped;

        bool IsRunning { get; }
        void Start(IEnumerable<IVsProject> projects);
        void RepeatLast();
        void StopIfRunning(bool blockUntilDone);
    }
}
